plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchMetadata.java

```diff
@@ -241,36 +249,14 @@ private Map<String, ColumnHandle> makeColumnHandles(List<IndexMetadata.Field> fi
             result.put(field.name(), new OpenSearchColumnHandle(
                     field.name(),
                     converted.type(),
+                    field.type(),
                     converted.decoderDescriptor(),
-                    supportsPredicates(field.type())));
+                    supportsPredicates(converted.type)));
         }
 
         return result.buildOrThrow();
     }
 
-    private static boolean supportsPredicates(IndexMetadata.Type type)
-    {
-        if (type instanceof DateTimeType) {
-            return true;
-        }
-
-        if (type instanceof PrimitiveType) {
-            switch (((PrimitiveType) type).name().toLowerCase(ENGLISH)) {
-                case "boolean":
-                case "byte":
-                case "short":
-                case "integer":
-                case "long":
-                case "double":
-                case "float":
-                case "keyword":
```

Comment:
Where is this being handled? I don't see it in the `supportsPredicates` method below.

Reply:
There is a RealType, DoubleType and VarcharType in `supportsPredicates`, is that what you mean?